src/views/postsFeed/PostsFeed.vue

```diff
@@ -0,0 +1,200 @@
+  },
+  watch: {
+    searchTerm() {
+      let term = this.$store.state.searchTerm;
+
+      if (this.topicId === "") {
+        this.fetchSearch({ query: term }).then(
+          data => {
+            this.displayedPosts = data.posts
+            this.$store.commit('setPosts', {posts: data.posts})
+          }
+        );
+      } else {
+        let id = this.topicId;
+        this.getTopicsInSearch({ topic: id, search: term }).then(
+          data => {
+            this.displayedPosts = data.posts
+            this.$store.commit('setPosts', {posts: data.posts})
+          }
+        );
+      }
+    },
+    $route(to, from) {
+      if (to.name === 'PostsAll') return this.fetchPosts()
+      if (to.name === 'Posts') return this.fetchTopicPosts()
+    }
+  },
+  // beforeRouteEnter(to, from, next) {
+  //   const topicSlug = to.params.slug;
+  //   if (topicSlug === undefined) {
+  //     return next(vm => {
+  //       if(vm.$store.state.searchTerm === null) {
+  //         vm.routerTopic = false
+  //         vm.fetchPosts()
+  //       }
+  //     });
+  //   }
+  //   next(vm =>
+  //     vm.$store
+  //       .dispatch("showTopic", topicSlug)
+  //       .then(topics => {
+  //         vm.routerTopic = true
+  //         vm.displayedPosts = topics.data.posts;
+  //         vm.$store.commit('setPosts', {posts: topics.data.posts})
+  //         vm.topicId = topics.data.topic[0]._id
+  //       })
+  //   );
+  // },
+  // beforeRouteUpdate(to, from, next) {
+  //   const topicSlug = to.params.topic;
+  //   this.$store.dispatch("showTopic", topicSlug).then(topics => {
+  //     this.routerTopic = false
+  //     this.endOfPosts = false
+  //     this.displayedPosts = topics.data.posts;
+  //     this.$store.commit('setPosts', {posts: topics.data.posts})
+  //   });
+  //   next();
+  // },
+  methods: {
+    ...mapActions(["showTopic", "getTopicsInSearch", "fetchSearch"]),
+    fetchPosts() {
+      this.topicId = ""
+      this.loading = true
+      this.getTopicsInSearch({}).then(
+        data => {
+          this.displayedPosts = data.posts
+          this.$store.commit('setPosts', {posts: data.posts})
+        }
+      ).finally(() => {
+        this.loading = false
+      })
+      this.resetPosts();
+    },
+    fetchTopicPosts() {
+      this.loading = true
+      this.$store.dispatch('showTopic', this.$route.params.slug).then((response) => {
+        if (response.data) {
+          this.displayedPosts = response.data.posts
+          if (response.data.topic) this.topicId = response.data.topic[0]._id
+        }
+      }).catch((e) => {
+        this.$toasted.error((e.response) ? e.response.data : e, { duration : 0 })
+      }).finally(() => {
+        this.loading = false
+      })
+    },
+    loadMore(newSearch = false) {
+      let isSearch = !!(this.$store.state.searchTerm)
+      let method = isSearch ? 'fetchSearch' : 'getTopicsInSearch'
+      if(this.routerTopic === true){
+        if(this.topicId){
```

Comment:
Do we need this nested conditional? `(this.routerTopic === true && this.topicId)`

Reply:
That part of the code was already done. I don't know exactly what it relates to. I prefer not to modify